src/main/java/com/commercetools/sync/services/impl/TaxCategoryServiceImpl.java

```diff
@@ -28,7 +30,10 @@ public TaxCategoryServiceImpl(@Nonnull final ProductSyncOptions syncOptions) {
 
     @Nonnull
     @Override
-    public CompletionStage<Optional<String>> fetchCachedTaxCategoryId(@Nonnull final String key) {
+    public CompletionStage<Optional<String>> fetchCachedTaxCategoryId(@Nullable final String key) {
+        if (isBlank(key)) {
+            return CompletableFuture.completedFuture(Optional.empty());
```

Comment:
`CompletableFuture.completedFuture(Optional.empty())`
we have 5 usage in the main sources + 11 usages in the tests.

Maybe it is time to make a static util method for this :) (but not so important, though).
Could it be a constant? (e.g. could you reused already completed/joined  future multiple times? guess yes, but not sure though)

Reply:
I agree, would be hard to make it constant though, since the types packed in the optionals are different. But it's worth investigating. I would say not in this PR though. An issue is created: https://github.com/commercetools/commercetools-sync-java/issues/164